src/GraphQL.Tests/Subscription/SubscriptionTests.cs

```diff
@@ -259,12 +259,7 @@ public async Task OnError()
 
             /* Then */
             var stream = result.Streams.Values.FirstOrDefault();
-            var message = await stream.FirstOrDefaultAsync();
-
-            message.ShouldNotBeNull();
-            message.ShouldBeOfType<ExecutionResult>();
-            message.Data.ShouldBeNull();
-            var error = message.Errors.Single();
+            var error = await Should.ThrowAsync<ExecutionError>(async () => await stream.FirstOrDefaultAsync());
```

Comment:
Why does it throw now?

Reply:
It raises an OnError event now, with an ExecutionError instance which is handled within https://github.com/graphql-dotnet/server/pull/754

Reply:
With PR 754 there is no net change.

Reply:
It does allow for sending an OnComplete message to the client notifying it that the source stream has triggered an OnError exception and no additional data is expected to be returned. But as of now it there is no net change.

Reply:
So in other words, with [server#754](https://github.com/graphql-dotnet/server/pull/754) merged in now, there is no net change, but it allows for [server#755](https://github.com/graphql-dotnet/server/pull/755) which we can discuss there.
- https://github.com/graphql-dotnet/server/pull/754
- https://github.com/graphql-dotnet/server/pull/755

I require this as I will be ensuring that my own implementation of the WebSockets handling code terminates the connection in the situation of a source stream error.

Reply:
OK. So Rx `FirstOrDefaultAsync` method just throws in case of `OnError`, right? It returned the whole `ExecutionResult` with error inside as data before.

Reply:
> So Rx FirstOrDefaultAsync method just throws in case of OnError, right?

Yes

> It returned the whole ExecutionResult with error inside as data before.

Yes

https://github.com/graphql-dotnet/server/pull/754 changes the server to support this change with no net change to the user -- it is simply transformed from an error to an ExecutionResult at another point in the process.

Reply:
I understood, thanks. My question was more about the nature of tests changes.